### PR TITLE
Show weather data based on given time of interest

### PR DESCRIPTION
**Pull request by Kyubinhan**
* Create a separate form component for requesting weather data with time of interest time picker
* Rename weather related variables
* Set PST in all the wx display

### CODE REVIEW

File: api/app/schemas/shared.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+"""
+
+from typing import List
+from pydantic import BaseModel
+
+
+class WeatherDataRequest(BaseModel):
+    """ A request for weather data for a given set of stations with a time of interest. """
+    stations: List[int]
+    time_of_interest: str = None
```
Comment (Sybrand) [inline api/app/schemas/shared.py:11]: How about defaulting to now as the time of interest? Then you don't need all that code checking for null and inserting it.
```suggestion
    time_of_interest: str = get_utc_now().isoformat()
```

File: api/app/main.py [hunk 2]
```diff
@@ -96,20 +98,29 @@ async def get_health():
...
 
 @api.post('/observations/', response_model=schemas.observations.WeatherStationHourlyReadingsResponse)
-async def get_hourlies(request: schemas.stations.StationCodeList, _: bool = Depends(authenticate)):
+async def get_hourlies(request: schemas.shared.WeatherDataRequest, _: bool = Depends(authenticate)):
     """ Returns hourly observations for the last 5 days, for the specified weather stations """
     try:
         logger.info('/observations/')
-        readings = await hourlies.get_hourly_readings(request.stations)
+
+        if request.time_of_interest is not None:
```
Comment (Sybrand) [inline api/app/main.py:117]: See my comment in WeatherDataRequest - I think we can get rid of the 4 lines of code that are repeated a few times.

**Review by Sybrand — CHANGES_REQUESTED** [review]
Could you look at moving get_utc_now into the schema? Also - I wonder about the "time of interest" label. Have you checked if that makes sense with Dana? Is he ok with that wording for "now" and for a point in the past?

File: api/app/schemas/shared.py (hunk 1 above)
Comment (Kyubinhan) [reply api/app/schemas/shared.py:11]: So the problem with this is that mocking `get_utc_now` doesn't work at all here. I couldn't figure out why so I went the non-elegant way. Thought?

File: web/src/utils/date.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import moment from 'moment'
 
-import { PST_UTC_OFFSET, PDT_UTC_OFFSET } from 'utils/constants'
+import { PST_UTC_OFFSET, PDT_UTC_OFFSET } from './constants'
```
Comment (andrea-williams) [inline web/src/utils/date.ts:3]: I thought we were always using absolute paths for imports?

File: web/src/features/fireWeather/components/graphs/plotlyHelper.ts [hunk 4]
```diff
@@ -379,18 +393,14 @@ export const getDailyAndAccumPrecips = (values: PrecipValue[]) => {
     })
 
     Object.entries(aggregatedPrecips).forEach(([formattedDate, totalPrecip]) => {
-      const midNightOfTheDay = moment(formattedDate)
-        .set({ hour: 0 })
-        .toDate()
+      const midNightOfTheDay = getMidnightDate(formattedDate)
```
Comment (andrea-williams) [inline web/src/features/fireWeather/components/graphs/plotlyHelper.ts:396]:
```suggestion
      const midnightOfTheDay = getMidnightDate(formattedDate)
```

File: web/src/features/fireWeather/components/graphs/plotlyHelper.ts [hunk 5]
```diff
@@ -379,18 +393,14 @@ export const getDailyAndAccumPrecips = (values: PrecipValue[]) => {
...
       dates.push(midNightOfTheDay)
       dailyPrecips.push(totalPrecip)
     })
   } else {
     values.forEach(({ datetime, total_precipitation }) => {
       if (total_precipitation !== undefined) {
-        const midNightOfTheDay = moment(datetime)
-          .set({ hour: 0 })
-          .toDate()
+        const midNightOfTheDay = getMidnightDate(datetime)
```
Comment (andrea-williams) [inline web/src/features/fireWeather/components/graphs/plotlyHelper.ts:403]:
```suggestion
        const midnightOfTheDay = getMidnightDate(datetime)
```

**Review by andrea-williams — APPROVED** [review]
Some minor changes requested, but otherwise I'm happy. Although I agree with Sy that it would be neater to set the timeOfInterest default value to now rather than having to check for None in a bunch of different places

**Review by Sybrand — APPROVED** [review]
I'm happy now - just check the "time of interest" wording with Dana/users.

File: web/src/utils/date.ts (hunk 3 above)
Comment (Kyubinhan) [reply web/src/utils/date.ts:3]: Yeah... but when Cypress references this file, it doesn't like absolute paths

**Pull request merged**